### PR TITLE
Delete resource even if it has non-appliedmanifestwork owners

### LINKED ISSUE #169: A removed configmap from the ManifestWork is not deleted in the ManagedCluster
I applied a manifestwork to my cluster which contains a configmap resource, and the configmap was created on the managed cluster successfully, after that I added another ownerreference to the configmap in the managed cluster. Then I deleted the manifestwork from the hub cluster, I expect the configmap should be deleted on the managed cluster, but it left.

1. Apply a manifestwork on the hub cluster.
```
apiVersion: work.open-cluster-management.io/v1
kind: ManifestWork
metadata:
  namespace: cluster1
  name: owner-test
spec:
  workload:
    manifests:
      - apiVersion: v1
        data:
          a: b
        kind: ConfigMap
        metadata:
          name: test1
          namespace: default
```
2. Add another ownerreference to the configmap on the managed cluster
```
apiVersion: v1
data:
  a: b
kind: ConfigMap
metadata:
  creationTimestamp: "2022-11-28T12:07:16Z"
  name: test1
  namespace: default
  ownerReferences:
  - apiVersion: work.open-cluster-management.io/v1
    kind: AppliedManifestWork
    name: 6348610e5141e7e2afafdbfb2519813b7e6ae0763763d58fe3f65f805628a132-owner-test
    uid: 46c8038e-3c6e-43e7-946a-c423b9d1543d
  - apiVersion: apps/v1
    kind: Deployment
    name: test-owner
    uid: 36cacd7f-1eb9-45cf-8626-b5f35f2f4841
  resourceVersion: "11153922"
  uid: 77eb7382-bfaa-4abd-91bd-736d9e5c6906
```
3. delete the manifestwork from hub.
4. check the configmap on the managed cluster, the AppliedManifestWork was removed, but the configmap itself was left.
```
apiVersion: v1
data:
  a: b
kind: ConfigMap
metadata:
  creationTimestamp: "2022-11-28T12:07:16Z"
  name: test1
  namespace: default
  ownerReferences:
  - apiVersion: apps/v1
    kind: Deployment
    name: test-owner
    uid: 36cacd7f-1eb9-45cf-8626-b5f35f2f4841
  resourceVersion: "11154294"
  uid: 77eb7382-bfaa-4abd-91bd-736d9e5c6906
```

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>
Fix: https://github.com/open-cluster-management-io/work/issues/169

### CODE REVIEW

File: test/e2e/work_agent_test.go [hunk 1]
```diff
@@ -626,6 +627,181 @@ var _ = ginkgo.Describe("Work agent", func() {
...
+			objects := []runtime.Object{
+				newConfigmap(nsName, cmName, nil, nil),
+			}
+			work2 := newManifestWork(clusterName, work2Name, objects...)
+			_, err := hubWorkClient.WorkV1().ManifestWorks(clusterName).Create(ctx, work2, metav1.CreateOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			for _, workName := range []string{workName, work2Name} {
+				// check status conditions in manifestwork status
+				gomega.Eventually(func() bool {
```
Comment (qiujian16) [inline test/e2e/work_agent_test.go]: return error here would make debugging easier.

File: test/e2e/work_agent_test.go [hunk 2]
```diff
@@ -626,6 +627,181 @@ var _ = ginkgo.Describe("Work agent", func() {
...
+
+			// wait for deletion of manifest work
+			gomega.Eventually(func() bool {
+				_, err := hubWorkClient.WorkV1().ManifestWorks(clusterName).Get(ctx, workName, metav1.GetOptions{})
+				return errors.IsNotFound(err)
+			}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
+
+			cm, err := spokeKubeClient.CoreV1().ConfigMaps(nsName).Get(ctx, cmName, metav1.GetOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			gomega.Expect(cm.CreationTimestamp.Equal(&cmCreationTime)).To(gomega.BeTrue())
```
Comment (qiujian16) [inline test/e2e/work_agent_test.go]: using uid to check might be clearer

**Comment by zhujian7** [comment]
@qiujian16 PTAL

**Pull request merged**